### DISCUSSION TITLE
i2s to i2s doesnt work - noise -  INMP441 to MAX98357a

### DISCUSSION THREAD

**Discussion by RenanEspinar** [pschatzmann/arduino-audio-tools#991] [Q&A] [answered]
Hello everyone.
I was trying to test some code from this project.
I realized that the code STREAMS I2S - I2S and STREAMS I2S - I2S 2 does not work.
This is my hardware:
- ESP32 - WROOM -32         ESP32S 32 PINS 
- INMP441               
     PIN 25;     -> WS 
     PIN 33;     -> SCK
     PIN 32;     -> SD               
- DFROBOT MAX98357a        
     PIN 14;   -> LRC
     PIN 15;   -> BCLK
     PIN 18;   -> DIN
ALSO my code is this:
`#include "AudioTools.h"
 
AudioInfo info(48000, 1, 16);
I2SStream in;
I2SStream out; 
StreamCopy copier(out, in); // copies sound into i2s

// Arduino Setup
void setup(void) {  
  // Open Serial 
  Serial.begin(115200);
  // change to Warning to improve the quality
  AudioLogger::instance().begin(Serial, AudioLogger::Info); 

  // start I2S in
  Serial.println("starting I2S...");
  auto config_in = in.defaultConfig(RX_MODE);
  config_in.copyFrom(info); 
  config_in.i2s_format = I2S_STD_FORMAT;
  config_in.is_master = true;
  config_in.port_no = 0;
  
  config_in.pin_ws =    25;     //WS 
  config_in.pin_bck =   33;     //SCK
  config_in.pin_data =  32;     //SD

  // config_in.fixed_mclk = sample_rate * 256
  // config_in.pin_mck = 2
  in.begin(config_in);

  // start I2S out
  auto config_out = out.defaultConfig(TX_MODE);
  config_out.copyFrom(info); 
  config_out.i2s_format = I2S_STD_FORMAT;
  config_out.is_master = true;
  config_out.port_no = 1;
  

  config_out.pin_ws =     14;   // LRC
  config_out.pin_bck =    15;   // BCLK
  config_out.pin_data =   18;   // DIN
  out.begin(config_out);

  Serial.println("I2S started...");
}

// Arduino loop - copy sound to out 
void loop() {
  copier.copy();
}`

**Comment by pschatzmann** [comment, marked as answer]
I am starting to get really annoyed! Search the issues and discussions first and read the documentation.
The issues with the INMP441 are properly documented!

https://github.com/pschatzmann/arduino-audio-tools#support
